smolyay/samples.py

```diff
@@ -508,3 +511,583 @@ def _create(self):
+
+    def __init__(self):
+        self._domain = None
+        self._points = None
+        self._valid_cache = False
+
+    @property
+    def num_dimensions(self):
+        """int: number of independent variables."""
+        return self.domain.shape[0]
+
+    @property
+    def domain(self):
+        """numpy.ndarray: domain of the point set."""
+        return self._domain
+
+    @property
+    def points(self):
+        """numpy.ndarray: Points to be sampled."""
+        if not self._valid_cache:
+            self._create()
+            self._valid_cache = True
+        return self._points
+
+    def __len__(self):
+        return self.points.shape[0]
+
+    @abc.abstractmethod
+    def _create(self):
+        """Abstract method for generating the mulitdimensional grid set"""
+        pass
+
+
+class RandomPointSet(MultidimensionalPointSet):
+    """Point Set for random points
+
+    Abstract class for multidimensional point sets that generate
+    random points using a seed.
+
+    Parameters
+    ----------
+    domain: list
+        domain of the random points
+
+    num_points: int
+        number of points to generate
+
+    seed: int
+        seed for generating the random points
+    """
+
+    def __init__(self, domain, num_points, seed):
+        super().__init__()
+        self._num_points = None
+        self._seed = None
+
+        self.domain = domain
+        self.num_points = num_points
+        self.seed = seed
+
+    @MultidimensionalPointSet.domain.setter
+    def domain(self, value):
+        domain = numpy.sort(numpy.array(value, ndmin=2), axis=1)
+        if domain.ndim != 2 or domain.shape[1] != 2:
+            raise TypeError("Domain must have size (num_dimensions, 2)")
+        if any(domain[:, 0] >= domain[:, 1]):
+            raise ValueError("Lower bound must be less than upper bound")
+        if not numpy.array_equal(self._domain, domain):
+            self._domain = domain
+            self._valid_cache = False
+
+    @property
+    def num_points(self):
+        """int: random seed for generating points"""
+        return self._num_points
+
+    @num_points.setter
+    def num_points(self, value):
+        num_points = int(value)
+        if self._num_points != num_points:
+            self._num_points = num_points
+            self._valid_cache = False
+
+    @property
+    def seed(self):
+        """int: random seed for generating points"""
+        return self._seed
+
+    @seed.setter
+    def seed(self, value):
+        value = int(value)
+        if self._seed != value:
+            self._seed = value
+            self._valid_cache = False
+
+
+class UniformRandomPointSet(RandomPointSet):
+    """Generates a grid using a uniform distribution"""
+
+    def _create(self):
+        """Generate a set of random points
+
+        Generates uniformly distributed random points
+        """
+        lower_bounds = [bound[0] for bound in self.domain]
+        upper_bounds = [bound[1] for bound in self.domain]
```

Comment:
These are NumPy arrays now, so you should be able to slice, right?
```suggestion
        lower_bounds = self.domain[:, 0]
        upper_bounds = self.domain[:, 1]
```